File: src/TypeFormatter/NullTypeFormatter.ts
import type { Definition } from "../Schema/Definition.js";
import type { SubTypeFormatter } from "../SubTypeFormatter.js";
import type { BaseType } from "../Type/BaseType.js";
import { NullType } from "../Type/NullType.js";

export class NullTypeFormatter implements SubTypeFormatter {
    public supportsType(type: BaseType): boolean {
        return type instanceof NullType;
    }
    public getDefinition(type: NullType): Definition {
        return { type: "null" };
    }
    public getChildren(type: NullType): BaseType[] {
        return [];
    }
}

File: internal/formatter/null_type_formatter.go
package formatter

import (
	"github.com/vega/ts-json-schema-generator/internal/schema"
	"github.com/vega/ts-json-schema-generator/internal/types"
)

// NullTypeFormatter mirrors src/TypeFormatter/NullTypeFormatter.ts.
type NullTypeFormatter struct{}

func NewNullTypeFormatter() *NullTypeFormatter { return &NullTypeFormatter{} }

func (f *NullTypeFormatter) SupportsType(t types.Type) bool {
	_, ok := t.(*types.NullType)
	return ok
}

func (f *NullTypeFormatter) GetDefinition(t types.Type) *schema.Definition {
	return &schema.Definition{Type: "null"}
}

func (f *NullTypeFormatter) GetChildren(t types.Type) []types.Type { return nil }
